python/cudf/cudf/dataframe/categorical.py

```diff
@@ -137,26 +138,30 @@ def __init__(self, **kwargs):
         self._categories = categories
         self._ordered = ordered
 
-    def serialize(self, serialize):
-        header, frames = super(CategoricalColumn, self).serialize(serialize)
+    def serialize(self):
+        header, frames = super(CategoricalColumn, self).serialize()
         header["ordered"] = self._ordered
-        header["categories"], category_frames = serialize(self._categories)
+        header["categories"], category_frames = self._categories.serialize()
         header["category_frame_count"] = len(category_frames)
+        header["type"] = pickle.dumps(type(self))
+        header["dtype"] = self._dtype.str
         frames.extend(category_frames)
         return header, frames
 
     @classmethod
-    def deserialize(cls, deserialize, header, frames):
-        data, mask = super(CategoricalColumn, cls).deserialize(
-            deserialize, header, frames
-        )
+    def deserialize(cls, header, frames):
+        data, mask = super(CategoricalColumn, cls).deserialize(header, frames)
+
         if "category_frame_count" not in header:
             # Handle data from before categories was a cudf.Column
             categories = header["categories"]
```

Comment:
Do we need this branch anymore? I think we can just remove this condition entirely.

Reply:
Removed